src/fsharp/FSharp.Build/Microsoft.FSharp.Targets

```diff
@@ -411,49 +410,72 @@ this file.
 
         <_CoreRelativeSuffix>.NETCore\$(TargetFSharpCoreVersion)\FSharp.Core.dll</_CoreRelativeSuffix>
         <_FrameworkRelativeSuffix>.NETFramework\v4.0\$(TargetFSharpCoreVersion)\FSharp.Core.dll</_FrameworkRelativeSuffix>
         <_PortableRelativeSuffix>.NETPortable\$(TargetFSharpCoreVersion)\FSharp.Core.dll</_PortableRelativeSuffix>
 
         <!-- .NETCore -->
         <_OldCoreSdkPath>$(_OldRootSdkLocation)\$(_CoreRelativeSuffix)</_OldCoreSdkPath>
-        <_NewCoreSdkPath>$(_NewRootSdkLocation)\$(_CoreRelativeSuffix)</_NewCoreSdkPath>
+        <_NewCoreSdkPath>$(NewFSharpSdkLocation)\$(_CoreRelativeSuffix)</_NewCoreSdkPath>
 
         <!-- .NETFramework\v4.0 -->
         <_OldFrameworkSdkPath>$(_OldRootSdkLocation)\$(_FrameworkRelativeSuffix)</_OldFrameworkSdkPath>
-        <_NewFrameworkSdkPath>$(_NewRootSdkLocation)\$(_FrameworkRelativeSuffix)</_NewFrameworkSdkPath>
+        <_NewFrameworkSdkPath>$(NewFSharpSdkLocation)\$(_FrameworkRelativeSuffix)</_NewFrameworkSdkPath>
 
         <!-- .NETPortable -->
         <_OldPortableSdkPath>$(_OldRootSdkLocation)\$(_PortableRelativeSuffix)</_OldPortableSdkPath>
-        <_NewPortableSdkPath>$(_NewRootSdkLocation)\$(_PortableRelativeSuffix)</_NewPortableSdkPath>
+        <_NewPortableSdkPath>$(NewFSharpSdkLocation)\$(_PortableRelativeSuffix)</_NewPortableSdkPath>
       </PropertyGroup>
 
       <ItemGroup>
         <!-- Update references to `.NETCore\*\FSharp.Core.dll`. -->
         <Reference Condition="'%(Reference.Identity)' == 'FSharp.Core' and
-                              '%(Reference.HintPath)' == '$(_OldCoreSdkPath)' and
-                              Exists('$(_NewCoreSdkPath)')">
+                              '%(Reference.HintPath)' == '$(_OldCoreSdkPath)'">
           <HintPath>$(_NewCoreSdkPath)</HintPath>
         </Reference>
 
         <!-- Update references to `.NETFramework\v4.0\*\FSharp.Core.dll`. -->
         <Reference Condition="'%(Reference.Identity)' == 'FSharp.Core' and
-                              '%(Reference.HintPath)' == '$(_OldFrameworkSdkPath)' and
-                              Exists('$(_NewFrameworkSdkPath)')">
+                              '%(Reference.HintPath)' == '$(_OldFrameworkSdkPath)'">
           <HintPath>$(_NewFrameworkSdkPath)</HintPath>
         </Reference>
 
         <!-- Update references to `.Portable\*\FSharp.Core.dll`. -->
         <Reference Condition="'%(Reference.Identity)' == 'FSharp.Core' and
-                              '%(Reference.HintPath)' == '$(_OldPortableSdkPath)' and
-                              Exists('$(_NewPortableSdkPath)')">
+                              '%(Reference.HintPath)' == '$(_OldPortableSdkPath)'">
           <HintPath>$(_NewPortableSdkPath)</HintPath>
         </Reference>
       </ItemGroup>
     </Target>
 
+
+    <Target Name="RedirectTPReferenceToNewRedistributableLocation" BeforeTargets="RedirectFSharpCoreReferenceToNewRedistributableLocation">
```

Comment:
Can `BeforeTargets` be changed to `ResolveAssemblyReferences`?  That way if the other target gets moved or renamed then this will still fire.